### DISCUSSION TITLE
Is there any way to close the ViewModel of the ContentControl

### DISCUSSION THREAD

**Discussion by zpccode** [canton7/Stylet#362] [Q&A] [answered]
After closing the parent page, the OnClose function of the child page is not called, and the instance of the child form is still in memory. How to clear it?

**Comment by canton7** [comment, marked as answer]
You probably want

```cs
TreatEffectSpotScanDetailsViewModel.ConductWith(this);
```

In the parent VM ctor (assuming that the parent isn't a Conductor, which you haven't shown). That way the child will know when the parent is activated/deactivated/closed.

That said, objects remaining in memory after they're no longer used is normal, and is how the .NET GC works.
